internal/serve/graphql/resolvers/mutations.resolvers.go

```diff
@@ -84,6 +91,110 @@ func (r *mutationResolver) DeregisterAccount(ctx context.Context, input graphql1
+	if err != nil {
+		return nil, &gqlerror.Error{
+			Message: fmt.Sprintf("Invalid operations: %s", err.Error()),
+			Extensions: map[string]interface{}{
+				"code": "INVALID_OPERATIONS",
+			},
+		}
+	}
+
+	// Convert simulation result if provided
+	var simulationResult entities.RPCSimulateTransactionResult
+	if transaction.SimulationResult != nil {
+		simulationResult = entities.RPCSimulateTransactionResult{
+			Events: transaction.SimulationResult.Events,
+		}
+
+		if transaction.SimulationResult.MinResourceFee != nil {
+			simulationResult.MinResourceFee = *transaction.SimulationResult.MinResourceFee
+		}
+		if transaction.SimulationResult.Error != nil {
+			simulationResult.Error = *transaction.SimulationResult.Error
+		}
+		if transaction.SimulationResult.LatestLedger != nil {
+			simulationResult.LatestLedger = int64(*transaction.SimulationResult.LatestLedger)
+		}
+
+		// Handle TransactionData if provided
+		if transaction.SimulationResult.TransactionData != nil {
+			var txData xdr.SorobanTransactionData
+			if txDataErr := xdr.SafeUnmarshalBase64(*transaction.SimulationResult.TransactionData, &txData); txDataErr != nil {
+				return nil, &gqlerror.Error{
+					Message: fmt.Sprintf("Invalid TransactionData: %s", txDataErr.Error()),
+					Extensions: map[string]interface{}{
+						"code": "INVALID_TRANSACTION_DATA",
+					},
+				}
+			}
+			simulationResult.TransactionData = txData
+		}
+	}
+
+	tx, err := r.transactionService.BuildAndSignTransactionWithChannelAccount(ctx, ops, int64(transaction.Timeout), simulationResult)
+	if err != nil {
+		switch {
+		case errors.Is(err, transactionservices.ErrInvalidArguments):
+			return nil, &gqlerror.Error{
+				Message: err.Error(),
+				Extensions: map[string]interface{}{
+					"code": "INVALID_ARGUMENTS",
+				},
+			}
+		case errors.Is(err, signing.ErrUnavailableChannelAccounts):
+			return nil, &gqlerror.Error{
+				Message: err.Error(),
+				Extensions: map[string]interface{}{
+					"code": "UNAVAILABLE_CHANNEL_ACCOUNTS",
+				},
+			}
+		case errors.Is(err, sorobanauth.ErrForbiddenSigner):
+			return nil, &gqlerror.Error{
+				Message: err.Error(),
+				Extensions: map[string]interface{}{
+					"code": "FORBIDDEN_SIGNER",
+				},
+			}
+		case errors.Is(err, store.ErrNoIdleChannelAccountAvailable):
+			return nil, &gqlerror.Error{
+				Message: err.Error(),
+				Extensions: map[string]interface{}{
+					"code": "NO_IDLE_CHANNEL_ACCOUNT",
+				},
+			}
+		default:
+			return nil, &gqlerror.Error{
+				Message: fmt.Sprintf("Failed to build transaction: %s", err.Error()),
+				Extensions: map[string]interface{}{
+					"code": "TRANSACTION_BUILD_FAILED",
+				},
+			}
+		}
+	}
+
+	// Convert transaction to XDR string
+	txXdrStr, err := tx.Base64()
+	if err != nil {
+		return nil, &gqlerror.Error{
+			Message: fmt.Sprintf("Failed to encode transaction to XDR: %s", err.Error()),
+			Extensions: map[string]interface{}{
+				"code": "XDR_ENCODING_FAILED",
+			},
+		}
+	}
```

Comment:
If we validated the transaction provided in the request, and our building logic is without bugs, then this shouldn't fail, so I don't know if we should handle the error here.